Modules/BenchPress.Azure/Private/Connect-Account.ps1

```diff
@@ -44,44 +47,70 @@ function Connect-Account {
   param ( )
   Begin { }
   Process {
-    $applicationId = Get-RequiredEnvironmentVariable AZ_APPLICATION_ID
-    $tenantId = Get-RequiredEnvironmentVariable AZ_TENANT_ID
+    $useManagedIdentity = Get-EnvironmentVariable AZ_USE_MANAGED_IDENTITY
     $subscriptionId = Get-RequiredEnvironmentVariable AZ_SUBSCRIPTION_ID
     $currentConnection = Get-AzContext
     $results = [AuthenticationResult]::new()
 
-    # If the current context matches the subscription, tenant, and service principal, then we're already properly
-    # logged in.
-    if ($null -ne $currentConnection `
-      -and ($currentConnection).Account.Type -eq 'ServicePrincipal' `
-      -and ($currentConnection).Account.Id -eq $applicationId `
-      -and ($currentConnection).Tenant.Id -eq $tenantId `
-      -and ($currentConnection).Subscription.Id -eq $subscriptionId) {
+      # Login Using Managed Identity
+    if ($useManagedIdentity) {
+      $connection = Connect-AzAccount -Identity
+      $subscriptionName = (Get-AzSubscription -SubscriptionId  $subscriptionId).Name
+      Set-AzContext -Subscription $subscriptionName
```

Comment:
After further testing, it only works with Id when using a User Assigned Managed Identity, with a System Assigned Managed Identity Set-AzContext throws an error unless the name is provided. This could be a bug in the azure-powershell sdk.